packages/flutter/lib/src/material/text_button.dart

```diff
@@ -377,18 +379,31 @@ class _TextButtonWithIconChild extends StatelessWidget {
     Key? key,
     required this.label,
     required this.icon,
+    required this.iconAlignment,
   }) : super(key: key);
 
   final Widget label;
   final Widget icon;
+  final ButtonIconAlignment iconAlignment;
 
   @override
   Widget build(BuildContext context) {
     final double scale = MediaQuery.maybeOf(context)?.textScaleFactor ?? 1;
     final double gap = scale <= 1 ? 8 : lerpDouble(8, 4, math.min(scale - 1, 1))!;
+    List<Widget> children;
+
+    switch (iconAlignment) {
+      case ButtonIconAlignment.start:
+        children = <Widget>[icon, SizedBox(width: gap), label];
+        break;
+      case ButtonIconAlignment.end:
+        children = <Widget>[label, SizedBox(width: gap), icon];
+        break;
+    }
```

Comment:
is there any way we can avoid duplicating this code everywhere?

Reply:
I don't know but from what I can see, buttons have a lot of duplications. I think it is okay because otherwise, we need to use a method that returns something. We could put it in `button_style.dart`.

Reply:
Using a method is a good way to reduce code duplication.

Reply:
ok sure, 🙈

Reply:
Done 🚀

Reply:
I have also made changes to the template because yesterday a pull request has been merged changing only `OutlinedButton.icon`
#70872